src/wallet/rpcwallet.cpp

```diff
@@ -1571,6 +1571,126 @@ UniValue listtransactions(const JSONRPCRequest& request)
+            "                                                      may be unknown for unconfirmed transactions not in the mempool\n"
+            "     \"details\" : [\n"
+            "       {\n"
+            "         \"account\" : \"accountname\",      (string) DEPRECATED. The account name involved in the transaction, can be \"\" for the default account.\n"
+            "         \"address\" : \"address\",          (string) The dogecoin address involved in the transaction\n"
+            "         \"category\" : \"send|receive\",    (string) The category, either 'send' or 'receive'\n"
+            "         \"amount\" : x.xxx,                 (numeric) The amount in " + CURRENCY_UNIT + "\n"
+            "         \"label\" : \"label\",              (string) A comment for the address/transaction, if any\n"
+            "         \"vout\" : n,                       (numeric) the vout value\n"
+            "         \"fee\": x.xxx,                     (numeric) The amount of the fee in " + CURRENCY_UNIT + ". This is negative and only available for the \n"
+            "                                              'send' category of transactions.\n"
+            "         \"abandoned\": xxx                  (bool) 'true' if the transaction has been abandoned (inputs are respendable). Only available for the \n"
+            "                                              'send' category of transactions.\n"
+            "       }\n"
+            "       ,...\n"
+            "     ],\n"
+            "     \"hex\" : \"data\"         (string) Raw data for transaction\n"
+            "   }\n"
+            "]\n"
+
+            "\nExamples:\n"
+            "\nList all transactions hashes in the wallet that do not have a mempool entry.\n"
+            + HelpExampleCli("liststucktransactions", "") +
+            "\nList all transactions hexs in the wallet that do not have a mempool entry \n"
+            + HelpExampleCli("liststucktransactions", "1") +
+            "\nAs a json rpc call\n"
+            + HelpExampleRpc("liststucktransactions", "")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    int verbosity = 0;
+    if (request.params.size() > 0)
+        verbosity = request.params[0].get_int();
+
+    isminefilter filter = ISMINE_SPENDABLE;
+    if(request.params.size() > 1)
+        if(request.params[1].get_bool())
+            filter = filter | ISMINE_WATCH_ONLY;
+
+    if(verbosity < 0 || verbosity > 2)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid verbosity");
+
+    UniValue ret(UniValue::VARR);
+
+    const CWallet::TxItems & txOrdered = pwalletMain->wtxOrdered;
+
+    for (CWallet::TxItems::const_iterator it = txOrdered.begin(); it != txOrdered.end(); ++it)
+    {
+        CWalletTx *const wtx = (*it).second.first;
+        if (wtx->GetDepthInMainChain() <= 0 && !wtx->InMempool() && wtx->IsFromMe(filter))
```

Comment:
Is this the right order for the conditionals? Because these comparisons short-circuit, we should figure out which of them is the cheapest and/or most likely to skip any unnecessary work.

My guess is `wtx->IsFromMe(filter)` is in the correct place, but I don't know for sure.

Reply:
Good point. Most of it will be memory operations. We have some bdb caching for the wallet entry (and we lock so it's extremely unlikely to cache-miss) and mempool tx hashes are always in-memory (as the name implies).  

I expect that:
1. `GetDepthInMainChain()` could be more expensive for old transactions than new ones - need to test
2. `InMempool()` is more expensive when mempool is full vs when it's empty, so this is a variable we do not control
3. `IsFromMe()` is arguably more expensive for large wallets than for small ones, but caching should negate that

I'll try to bench this against the reverse order we use here to find out - I have a very dirty 25MB testnet wallet, hope that's enough to see a difference

Reply:
That's plenty fast right now with a healthy number of transactions. If we see an order of magnitude increase in the number of transactions and this code scales linearly (yeah, I know about assumptions), that still seems like acceptable RPC performance. I'm fine with this unmodified then.